bindings/wallet-c/src/time.rs

```diff
@@ -0,0 +1,35 @@
+use wallet_core::c::time::{compute_end_date, BlockDate};
+
+use crate::{ErrorPtr, Settings};
+
+/// This function dereference raw pointers. Even though the function checks if
+/// the pointers are null. Mind not to put random values in or you may see
+/// unexpected behaviors.
+///
+/// # Arguments
+///
+/// *settings*: the blockchain settings previously allocated with this library.  
+/// *final_date*: desired date of expiration for a fragment. If 0 is passed then the expiration
+/// date will be set to the maximum possible date.
+/// *block_date_out*: pointer to an allocated BlockDate structure, the memory should be writable.
+///
+/// # Safety
+///
+/// pointers should be allocated by this library and be valid.
+/// null pointers are checked and will result in an error.
+///
+#[no_mangle]
+pub unsafe extern "C" fn iohk_jormungandr_compute_ttl_from_date(
```

Comment:
Let's be consistent: use something like `block_date_from_system_time` instead of various `compute_*_from_date` everywhere because this is what it essentially does. We may have more uses for this conversion than the expiry date.

Reply:
I agree with the name, but this function is also currently giving you the maximum expiration date if you pass 0 to it. Which I guess should be extracted to a separate function.

Reply:
What I'm not really sure about is how would you validate the expiration date. Should we make it optional like (in rust for practicality):

```rust
let now = SystemTime::now();
let block_date = block_date_from_system_time(settings, now + N);
let max_date = max_expiration_date(settings, now);
if (block_date <= max_date) {
    // wallet.vote...
}
else {
    // I don't know, reduce N by some amount I guess
}
```

I was going to check it in the same function that the user calls ( block_date_from_system_time), but to validate you need to calculate what's the maximum, so I did both at once... But it's hard to test, so splitting it like that seems like the simplest way of doing it?

Reply:
Although, we could also skip computing the max and validating for another time.